WordPress/gutenberg
# (Custom)SelectControl: Refresh and refactor chevron

Follow-up to #42460

## What?

Refactors the `chevronDown` icon in SelectControl/CustomSelectControl so the implementation is clean and the visual is consistent.

### Visual changes

- Paddings are now dynamic based on the size variant.

@jasmussen Are we happy with the size of the icon? And the spacing before and after it? Let me know if you have tweaks.

## Why?

We now have a clean way to implement the chevron (`suffix` prop & suffix wrapper component). It removes hacks and maintains visual consistency across components.

## How?

Refactor the implementation to use the `suffix` prop & suffix wrapper component.

## Testing Instructions

1. `npm run storybook:dev`
2. See the stories for SelectControl and CustomSelectControl. Check the size variants.

✅ Works in RTL
✅ Width behavior is maintained for CustomSelectControl. (130px wide by default, and grows when the label is long)

## Screenshots or screencast 

Here's a video of all the size variants (24, 30, 36, 40):

https://user-images.githubusercontent.com/555336/182780159-c407e2a2-56bc-4d5a-9348-78e51838bb7a.mp4
